### ISSUE TITLE
Regression with 3.0.8: loosing extensions

### ISSUE THREAD

**Issue by kylekatarnls** [phpseclib/phpseclib#1642]
Hello 👋 

I've an issue running the following code:

```php
$issuerKey = phpseclib3\Crypt\EC::createKey('ed25519');
$subjectKey = phpseclib3\Crypt\EC::createKey('ed25519')->getPublicKey();

$subject = new X509();
$subject->setPublicKey($subjectKey);
$subject->setDN(['commonName' => 'subject']);

$issuer = new X509();
$issuer->setPrivateKey($issuerKey);
$issuer->setDN(['commonName' => 'issuer']);

$authority = new X509();

$authority->setExtensionValue('id-ce-keyUsage', ['digitalSignature']);

$cert = $authority->saveX509($authority->sign($issuer, $subject));

$loader = new X509();

var_dump($loader->loadX509($cert)['tbsCertificate']['extensions']);
```

Before [this commit](https://github.com/phpseclib/phpseclib/commit/701881d3de21cb5e96a5d2bac82e0615f3a1bda6) (and 3.0.7 or with the branch of#1640), I got:
```array(1) {
  [0] =>
  array(3) {
    'extnId' =>
    string(14) "id-ce-keyUsage"
    'critical' =>
    bool(false)
    'extnValue' =>
    array(1) {
      [0] =>
      string(16) "digitalSignature"
    }
  }
}
```

After this commit, I get:
```
Notice: Undefined index: extensions
NULL
```

I suspect `$extensions = &$this->subArray($root, $path)` not seeding properly by reference `$root['tbsCertificate']['extensions']`.

Thanks.

**Comment by kylekatarnls** [comment]
For the record, I also tried the `$replace = true` new flag but no better luck:

```
Warning: Invalid argument supplied for foreach()
```

Actually at this point `$extensions` is `false` as `$create` is disabled, so this return statement applies:
```php
if (!$create) {
    return $false;
}
```

**Comment by kylekatarnls** [comment]
I proposed a fix #1643 and a regression test to secure it, waiting for Travis but locally all tests still pass.

Side note is it intended that `setExtensionValue()` does nothing using `$replace = true` if no value was previously set?

**Comment by terrafrost** [comment]
#1643 has been merged.

Thanks!
